File: incomplete_fizz_buzz/java/src/FizzBuzzRunner.java
public class FizzBuzzRunner {
  private Inputable inputer;
  private Printable printer;
  private History history = new History();

  public FizzBuzzRunner(Inputable inputer, Printable printer) {
    this.inputer = inputer;
    this.printer = printer;
  }

  public void run(String selector) {
    switch(selector) {
      case "1":
        int number = Integer.parseInt(inputer.gets());
        history.add(number);
        FizzBuzz.fizzBuzz(number, printer);
        break;
      case "2":
        history.print(printer);
        break;
      case "3":
        File.write(history);
        break;
      case "4":
        File.read(printer);
    }
  }
}

File: incomplete_fizz_buzz/php/lib/FizzBuzzRunner.php
<?php

require_once('FizzBuzz.php');

class FizzBuzzRunner
{

  public function __construct($inputer, $printer)
  {
    $this->inputer = $inputer;
    $this->printer = $printer;
    $this->histories = [];
  }

  public function run($selector)
  {
    if($selector === '1') {
      $number =  intval($this->inputer->gets());
      $this->histories[] = $number;
      FizzBuzz::fizz_buzz($number, $this->printer);
    } else if($selector === '2') {
      for($i = 0; $i < count($this->histories); $i++) {
        FizzBuzz::fizz_buzz_history($this->histories[$i], $this->printer);
      }
    } else if($selector === '3') {
      FizzBuzz::write($this->histories);
    } else if($selector === '4') {
      if (file_exists('data.txt')) {
        $file = fopen('data.txt', 'r');
        while(true) {
          $line = fgets($file);
          if($line === false){
            break;
          }
          $this->printer->execute(trim($line));
        }
        fclose($file);
      }
    }
  }
}
